fmriprep/workflows/bold/confounds.py

```diff
@@ -586,18 +612,67 @@ def _getbtthresh(medianval):
         (melodic, ica_aroma, [('out_dir', 'melodic_dir')]),
         # generate tsvs from ICA-AROMA
         (ica_aroma, ica_aroma_confound_extraction, [('out_dir', 'in_directory')]),
+        (inputnode, ica_aroma_confound_extraction, [
+            ('skip_vols', 'skip_vols')]),
         # output for processing and reporting
         (ica_aroma_confound_extraction, outputnode, [('aroma_confounds', 'aroma_confounds'),
                                                      ('aroma_noise_ics', 'aroma_noise_ics'),
                                                      ('melodic_mix', 'melodic_mix')]),
         # TODO change melodic report to reflect noise and non-noise components
-        (ica_aroma, outputnode, [('nonaggr_denoised_file', 'nonaggr_denoised_file')]),
+        (ica_aroma, add_non_steady_state, [
+            ('nonaggr_denoised_file', 'bold_cut_file')]),
+        (bold_mni_trans_wf, add_non_steady_state, [
+            ('outputnode.bold_mni', 'bold_file')]),
+        (inputnode, add_non_steady_state, [
+            ('skip_vols', 'skip_vols')]),
+        (add_non_steady_state, outputnode, [('bold_add', 'nonaggr_denoised_file')]),
         (ica_aroma, ds_report_ica_aroma, [('out_report', 'in_file')]),
     ])
 
     return workflow
 
 
+def _remove_volumes(bold_file, skip_vols):
+        import nibabel as nb
+        from nipype.utils.filemanip import fname_presuffix
+
+        # load the bold file and get the 4d matrix
+        bold_img = nb.load(bold_file)
+        bold_data = bold_img.get_data()
+
+        # cut off the beginning volumes
+        bold_data_cut = bold_data[..., skip_vols:]
+
+        # modify header with new shape (fewer volumes)
+        data_shape = list(bold_img.header.get_data_shape())
+        data_shape[-1] -= skip_vols
+        bold_img.header.set_data_shape(tuple(data_shape))
+
+        # save the resulting bold file
+        out = fname_presuffix(bold_file, suffix='_cut')
+        bold_img.__class__(bold_data_cut, bold_img.affine, bold_img.header).to_filename(out)
+        return out
+
+
+def _add_volumes(bold_file, bold_cut_file, skip_vols):
+    """prepend skip_vols from bold_file onto bold_cut_file"""
+    import nibabel as nb
+    from nipype.utils.filemanip import fname_presuffix
+
+    # load the data
+    bold_img = nb.load(bold_file)
+    bold_data = bold_img.get_data()
+    bold_cut_img = nb.load(bold_cut_file)
+    bold_cut_data = bold_cut_img.get_data()
+
+    # assign everything from skip_vols foward to bold_cut_data
+    bold_data[..., skip_vols:] = bold_cut_data
```

Comment:
What about:

```Python
bold_img = nb.load(bold_file)
bold_cut_img = nb.load(bold_cut_file)

bold_data = np.concatenate((bold_img.dataobj[..., :skip_vols],
                            bold_cut_img.dataobj), axis=3)
```

This would reduce the memory usage by the uncompressed size of `bold_cut_img`.

Or, if we're willing to require `nibabel>=2.3`:

```Python
new_img = nb.concat_images((bold_img.slicer[..., :skip_vols], bold_cut_img), axis=3)
```

This last has the advantage of checking affines to ensure the images are in the same space.